gossip/src/duplicate_shred.rs

```diff
@@ -219,17 +220,20 @@ fn check_chunk(
         } else if dup.num_chunks != num_chunks {
             Err(Error::NumChunksMismatch)
         } else if dup.chunk_index >= num_chunks {
-            Err(Error::InvalidChunkIndex)
+            Err(Error::InvalidChunkIndex {
+                chunk_index: dup.chunk_index,
+                num_chunks,
+            })
         } else {
             Ok(())
         }
     }
 }
 
 /// Reconstructs the duplicate shreds from chunks of DuplicateShred.
-pub fn into_shreds(
+pub(crate) fn into_shreds(
+    pubkey: &Pubkey, // slot leader
```

Comment:
Why not rename pubkey here to be just slot_leader?

Reply:
renamed to `slot_leader`